internal/hook/item_hook_handler.go

```diff
@@ -407,3 +411,106 @@ func GetRestoreHooksFromSpec(hooksSpec *velerov1api.RestoreHooks) ([]ResourceRes
 
 	return restoreHooks, nil
 }
+
+// getPodExecRestoreHookFromAnnotations returns an ExecRestoreHook based on restore annotations, as
+// long as the 'command' annotation is present. If it is absent, this returns nil.
+func getPodExecRestoreHookFromAnnotations(annotations map[string]string, log logrus.FieldLogger) *velerov1api.ExecRestoreHook {
+	commandValue := annotations[podRestoreHookCommandAnnotationKey]
+	if commandValue == "" {
+		return nil
+	}
+
+	container := annotations[podRestoreHookContainerAnnotationKey]
+
+	onError := velerov1api.HookErrorMode(annotations[podRestoreHookOnErrorAnnotationKey])
+	if onError != velerov1api.HookErrorModeContinue && onError != velerov1api.HookErrorModeFail {
+		onError = ""
```

Comment:
That does seem like a better default but then it would differ from how Backup exec hooks work. I've left it unchanged for now pending further discussion.

Reply:
Unresolved this for discussion 👇 

Seems like the `hookLog` defaults the `onError` to `Fail`
https://github.com/vmware-tanzu/velero/blob/main/pkg/podexec/pod_command_executor.go#L103-L109
The fact that the log message says `OnError=Fail` but the restore completing successfully kinda seems like saying one thing but doing something else.

suggest using `Fail` as the default here for restore hooks and opening an issue to do the same thing for the backup hooks as well.
WDYT? @nrb

Reply:
this can be addressed in a follow-up